pubsub/rabbitpubsub/rabbit.go

```diff
@@ -580,6 +580,11 @@ func (s *subscription) ReceiveBatch(ctx context.Context, maxMessages int) ([]*dr
 				}
 				// s.closec must be closed too. See if it has an error.
 				if err := closeErr(s.closec); err != nil {
+					// PreconditionFailed can happen if we send an Ack or Nack for a
```

Comment:
@jba This is also super sucky. It looks like when we send acks/nacks, errors are reported back by closing the channel and we get them the next time we call `Receive`; all we can do is reopen it. Suggestions are welcome.

Reply:
When we do the ack, could we check for channel closure there by selecting from s.closec?

Reply:
No; I tried adding a `select` on `s.closec` at the end of `sendAcksOrNacks`, but there was never any error there, even when one later shows up. There must be some delay in closing the channel/sending the error.